source/common/buffer/buffer_impl.cc

```diff
@@ -44,6 +44,16 @@ void OwnedImpl::add(const Instance& data) {
 }
 
 void OwnedImpl::prepend(absl::string_view data) {
+  // Prepending an empty string seems to mess up libevent internally.
```

Comment:
Is it worth opening an upstream issue on this and linking here?

Reply:
I'll take an AI to work on a minimal repro using just libevent calls and file on their GH.

Reply:
See https://github.com/libevent/libevent/issues/774